kernel/satgen.cc

```diff
@@ -430,15 +430,15 @@ bool SatGen::importCell(RTLIL::Cell *cell, int timestep)
 		return true;
 	}
 
-	if (cell->type.in(ID($pos), ID($buf), ID($neg)))
+	if (cell->type.in(ID($pos), ID($buf), ID($barrier), ID($neg)))
```

Comment:
I think by supplying a SAT model for `$barrier` to the optimization passes you are running the risk of barrier leakage. I would make the model opt-in, not sure how to do it best though